### PR TITLE
ui: Add deny SVG lines with icons

### PR DESCRIPTION
**Pull request by kaxcode**
Use Cases:
1. If intention is set to deny between services, the SVG lines is red with an icon.
2. If intention is set with L7, the line remains gray but has a L7 icon.

New contextual components:
- TopologyMetrics::Icon
- TopologyMetrics::UpLines
- TopologyMetrics::DownLines

Screen Shot 2020-10-06 at 7 25 53 PM

### CODE REVIEW

File: ui-v2/app/components/topology-metrics/index.js [hunk 1]
```diff
@@ -12,41 +12,65 @@ export default class TopologyMetrics extends Component {
...
-      const dimensions = item.getBoundingClientRect();
-      const dest = {
-        x: this.centerDimensions.x,
-        y: this.centerDimensions.y + this.centerDimensions.height / 4,
-      };
-      const src = {
-        x: dimensions.x + dimensions.width,
-        y: dimensions.y + dimensions.height / 2,
-      };
+    let order = [null, 'allow', 'deny'];
```
Comment (johncowen) [inline ui-v2/app/components/topology-metrics/index.js:15]: A random `let` makes an appearance! 😂  You usually use `const` everywhere (where you aren't changing the reference), should this be `const` also?

File: ui-v2/app/components/topology-metrics/index.js [hunk 2]
```diff
@@ -12,41 +12,65 @@ export default class TopologyMetrics extends Component {
...
-        src: src,
-      };
-    });
+    return items
+      .map(item => {
+        const dimensions = item.getBoundingClientRect();
+        const dest = {
+          x: this.centerDimensions.x,
+          y: this.centerDimensions.y + this.centerDimensions.height / 4,
+        };
```
Comment (johncowen) [inline ui-v2/app/components/topology-metrics/index.js:23]: Just noticed you could move this out of the `map` above the `return items`, it stays the same for every iteration of the loop.

File: ui-v2/app/components/topology-metrics/index.js [hunk 3]
```diff
@@ -12,41 +12,65 @@ export default class TopologyMetrics extends Component {
...
+          dest: dest,
+          src: src,
+        };
+      })
+      .sort((a, b) => {
+        if (a.permission === b.permission) {
+          return 0;
+        }
+
+        return order.indexOf(a.permission) - order.indexOf(b.permission);
```
Comment (johncowen) [inline ui-v2/app/components/topology-metrics/index.js:41]: Nice! Actually, do you even need the above conditional now? If they are both allow then `1 - 1 = 0`, both deny `2 - 2 = 0`

File: ui-v2/app/components/topology-metrics/index.js [hunk 4]
```diff
@@ -12,41 +12,65 @@ export default class TopologyMetrics extends Component {
...
+      .map(item => {
+        const dimensions = item.getBoundingClientRect();
+        const dest = {
+          x: dimensions.x - dimensions.width - 26,
+          y: dimensions.y + dimensions.height / 2,
+        };
+        const src = {
+          x: this.centerDimensions.x + 20,
+          y: this.centerDimensions.y + this.centerDimensions.height / 4,
+        };
```
Comment (johncowen) [inline ui-v2/app/components/topology-metrics/index.js:58]: Here you can hoist this one out of the map iteration, similar to `dest` above.

**Review by johncowen — COMMENTED** [review]
This is looking super good! 👏 

I noticed we need to make sure the icons change position when the data changes also (so blocking queries basically) - guessing we can probably do similar to what we originally did to get the lines updating for blocking queries?

I left a couple more non-important inline comments seeing, if you feel like changing those while you are in there then great, no prob if not.

File: ui-v2/app/components/topology-metrics/index.js (hunk 1 above)
Comment (kaxcode) [reply ui-v2/app/components/topology-metrics/index.js:15]: 😂 no clue why I thought I'd need them

File: ui-v2/app/components/topology-metrics/down-lines/index.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+export default class TopoloyMetricsLines extends Component {
```
Comment (johncowen) [inline ui-v2/app/components/topology-metrics/down-lines/index.js:5]: Doesn't matter - but theres a typo here if you want to fix that up?

File: ui-v2/app/components/topology-metrics/up-lines/index.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+export default class TopoloyMetricsLines extends Component {
+  @tracked iconPositions;
```
Comment (johncowen) [inline ui-v2/app/components/topology-metrics/up-lines/index.js:6]: Theres that typo again!

**Review by johncowen — APPROVED** [review]
LGTM! I spotted a couple of non-important typos you may wanna fix up whilst you are waiting on backend support

**Pull request merged**